plugins/systemstat.koplugin/main.lua

```diff
@@ -0,0 +1,79 @@
+
+local KeyValuePage = require("ui/widget/keyvaluepage")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local _ = require("gettext")
+
+local SystemStat = {
+    start_sec = os.time(),
+    wakeup_count = 0,
+    sleep_count = 0,
+    charging_count = 0,
+    discharging_count = 0,
+}
+
+function SystemStat:onSuspend()
+    self.sleep_count = self.sleep_count + 1
+end
+
+function SystemStat:onResume()
+    self.wakeup_count = self.wakeup_count + 1
+end
+
+function SystemStat:onCharging()
+    self.charging_count = self.charging_count + 1
+end
+
+function SystemStat:onNotCharging()
+    self.discharging_count = self.discharging_count + 1
+end
+
+function SystemStat:onCallback()
```

Comment:
nitpick, i suggest renaming `onCallback` to something like `showStatistics` for two reasons:

1. The coding convention is to use `onFoo` to define a handler for event `Foo`. In this case, there is no `Callback` event.
2. `onCallback` is such a generic name that it's hard to tell what this function actually does without looking at the implementation.